x/ccv/provider/keeper/partial_set_security_test.go

```diff
@@ -683,3 +683,122 @@ func findConsumerValidator(t *testing.T, v types.ConsumerValidator, valsAfter []
+	}
+
+	return validators, consAddrs
+}
+
+// TestMinStake checks that FulfillsMinStake returns true if the validator has more than the min stake
+// and false otherwise
+func TestMinStake(t *testing.T) {
+	providerKeeper, ctx, ctrl, mocks := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+	defer ctrl.Finish()
+
+	// create two validators with powers 1 and 2
+	_, consAddrs := createStakingValidatorsAndMocks(ctx, mocks, 1, 2)
+
+	testCases := []struct {
+		name           string
+		minStake       uint64
+		expectedFulfil []bool
+	}{
+		{
+			name:           "No min stake",
+			minStake:       0,
+			expectedFulfil: []bool{true, true},
+		},
+		{
+			name:           "Min stake set to 2",
+			minStake:       2,
+			expectedFulfil: []bool{false, true},
+		},
+		{
+			name:           "Min stake set to 3",
+			minStake:       3,
+			expectedFulfil: []bool{false, false},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			providerKeeper.SetMinStake(ctx, "chainID", tc.minStake)
+			for i, valAddr := range consAddrs {
+				result := providerKeeper.FulfillsMinStake(ctx, "chainID", valAddr)
+				require.Equal(t, tc.expectedFulfil[i], result)
+			}
+		})
+	}
+}
+
+// TestMaxProviderConsensusValidators checks that the number of validators in the next validator set is at most
+// the maxProviderConsensusValidators parameter if the consumer chain does not allow inactive validators to validate.
+func TestIfInactiveValsDisallowedProperty(t *testing.T) {
+	rapid.Check(t, func(r *rapid.T) {
+		providerKeeper, ctx, ctrl, mocks := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+		defer ctrl.Finish()
+
+		// Generate a random number of validators with random powers
+		valPowers := rapid.SliceOfN(rapid.Int64Range(1, 100), 1, 100).Draw(r, "valPowers")
+		vals, consAddrs := createStakingValidatorsAndMocks(ctx, mocks, valPowers...)
+
+		// opt the validators in
+		for _, valAddr := range consAddrs {
+			providerKeeper.SetOptedIn(ctx, "chainID", valAddr)
+		}
+
+		// Randomly choose values for parameters
+		minStake := rapid.Uint64Range(0, 101).Draw(r, "minStake")
+		maxProviderConsensusVals := rapid.Uint32Range(1, 11).Draw(r, "maxProviderConsensusVals")
+
+		// Set up the parameters in the provider keeper
+		providerKeeper.SetAllowInactiveValidators(ctx, "chainID", false) // do not allow inactive validators
+		providerKeeper.SetMinStake(ctx, "chainID", minStake)
+		params := providerKeeper.GetParams(ctx)
+		params.MaxProviderConsensusValidators = int64(maxProviderConsensusVals)
+		providerKeeper.SetParams(ctx, params)
+
+		// Compute the next validators
+		nextVals := providerKeeper.ComputeNextValidators(ctx, "chainID", vals)
+
+		// Check that the length of nextVals is at most maxProviderConsensusVals
+		require.LessOrEqual(r, len(nextVals), int(maxProviderConsensusVals), "The length of nextVals should be at most maxProviderConsensusVals")
```

Comment:
I guess those might not be equal if we have less randomly chosen `valPowers` validators than the randomly chosen `maxProviderConsensusVals`?